course6/toyapp_thomasdhuys/drawlib/matrix.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <array>
+
+#include "vertex.h"
+
+namespace Draw {
+
+namespace {
+	static constexpr size_t COLS = 4;
+	static constexpr size_t ROWS = 4;
+}
+
+using RowType = std::array<VCoord, COLS>;
+
+class Matrix4x4 {
```

Comment:
I was hoping that now your Matrix would be a template, with both size and the element type being template parameters.